public/app/plugins/datasource/prometheus/querybuilder/components/PromQueryBuilder.tsx

```diff
@@ -42,11 +42,85 @@ export const PromQueryBuilder = React.memo<Props>((props) => {
   const { datasource, query, onChange, onRunQuery, data, showExplain } = props;
   const [highlightedOp, setHighlightedOp] = useState<QueryBuilderOperation | undefined>();
   const [showDrawer, setShowDrawer] = useState<boolean>(false);
+  const [llmAppEnabled, updateLlmAppEnabled] = useState<boolean>(false);
 
   const lang = { grammar: promqlGrammar, name: 'promql' };
 
   const initHints = datasource.getInitHints();
 
+  useEffect(() => {
+    async function checkLlms() {
+      const check = (await llms.openai.enabled()).ok;
```

Comment:
This should be checking `llms.openai` and `llms.vector`, since we need both for this feature.

Reply:
Might be nice to remove the ok check in `helper > promQailSuggest` since we don't need that check anymore.

Reply:
I there's an issue with forcing the check to have the vector too, the docs don't talk about vector
https://grafana.com/docs/grafana-cloud/alerting-and-irm/machine-learning/llm-plugin/